packages/ember-views/lib/system/build-component-template.js

```diff
@@ -29,17 +24,13 @@ export default function buildComponentTemplate(componentInfo, attrs, content) {
       var attributes = normalizeComponentAttributes(component, attrs);
       var elementTemplate = internal.manualElement(tagName, attributes);
 
-      createdElementBlock = true;
-
       blockToRender = createElementBlock(elementTemplate, blockToRender, component);
     } else {
       validateTaglessComponent(component);
     }
-
-    return { createdElement: tagName !== '', block: blockToRender };
   }
 
-  return { createdElement: false, block: blockToRender };
+  return { createdElement: !!tagName, block: blockToRender };
```

Comment:
If you are back in this code, can you add a comment before this line? `tagName` is tricky and though the logic here is correct it is pretty obtuse. Something like:

```js
// tagName is one of:
//
//   * `undefined` if no component is present
//   * the falsy value "" if set explicitly on the component
//   * an actual tagName set explicitly on the component
```